app/v3/validators/resolvers/ResolvePeriodId.scala

```diff
@@ -49,3 +49,20 @@ object ResolvePeriodId {
     }
 
 }
+
+object ResolveTysTaxYearWithMax extends ResolverSupport {
```

Comment:
Please remove ResolveTysTaxYearWithMax from here (in both v3 and v4) and use ResolveTaxYearMinMax from the shared code.